test/cpp/asyncprogressworkerstream.cpp

```diff
@@ -0,0 +1,83 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2016 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#ifndef _WIN32
+#include <unistd.h>
+#define Sleep(x) usleep((x)*1000)
+#endif
+#include <nan.h>
+
+using namespace Nan;  // NOLINT(build/namespaces)
+
+// custom data type. This servers as an example of how external
+// libraries could be hooked in, populate their objects and send them to JS
+struct data_t {
+  int index;
+  int data;
+};
+
+// unlike test ayyncprogressworker.cpp this test is explictely templated
+template<typename T>
+class ProgressWorker : public AsyncProgressWorker<T> {
+ public:
+  ProgressWorker(
+      Callback *callback
+    , Callback *progress
+    , int milliseconds
+    , int iters)
+    : AsyncProgressWorker<T>(callback), progress(progress)
+    , milliseconds(milliseconds), iters(iters) {}
+  ~ProgressWorker() {}
+
+  void Execute (const typename AsyncProgressWorker<T>::ExecutionProgress& progress) {
+    data_t data;
+    for (int i = 0; i < iters; ++i) {
+      data.index = i;
+      data.data = i * 2;
+      progress.Send(&data, sizeof(data_t));
```

Comment:
with space, it is a keyword, not a function

On June 30, 2016 5:19:49 PM GMT+03:00, Ben Noordhuis <notifications@github.com> wrote:
>> + public:
>> +  ProgressWorker(
>> +      Callback *callback
>> +    , Callback *progress
>> +    , int milliseconds
>> +    , int iters)
>> +    : AsyncProgressWorker<T>(callback), progress(progress)
>> +    , milliseconds(milliseconds), iters(iters) {}
>> +  ~ProgressWorker() {}
>> +
>> +  void Execute (const typename
>AsyncProgressWorker<T>::ExecutionProgress& progress) {
>> +    data_t data;
>> +    for (int i = 0; i < iters; ++i) {
>> +      data.index = i;
>> +      data.data = i * 2;
>> +      progress.Send(&data, sizeof(data_t));
>
>`sizeof(data)`
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/588/files/85aa1f0be5361cd79a2c85f75cb6296188223c87#r69140490